api/client/src/main/java/jakarta/websocket/RemoteEndpoint.java

```diff
@@ -275,16 +275,23 @@
 
         /**
          * Opens an output stream on which a binary message may be sent. The developer must close the output stream in
-         * order to indicate that the complete message has been placed into the output stream.
+         * order to indicate that the complete message has been placed into the output stream. If no calls are made to
+         * one of the {@code write()} methods before the output stream is closed then no WebSocket binary messages will
+         * be sent. If at least one call is made to one of the {@code write()} methods before the output stream is
+         * closed then at least one WebSocket binary message will be sent even if that message is of zero length.
          *
          * @return the output stream to which the message will be written.
          * @throws IOException if there is a problem obtaining the OutputStream to write the binary message.
          */
         OutputStream getSendStream() throws IOException;
 
         /**
-         * Opens an character stream on which a text message may be sent. The developer must close the writer in order
-         * to indicate that the complete message has been placed into the character stream.
+         * Opens a character stream on which a text message may be sent. The developer must close the writer in order to
+         * indicate that the complete message has been placed into the character stream. If no calls are made to one of
+         * the {@code write()} or {@code append)} methods before the writer is closed then no WebSocket text messages
+         * will be sent. If at least one call is made to one of the {@code write()} or {@code append)} methods before
```

Comment:
I think you dropped `(` here (and the line above).

Reply:
Thanks I'll fix that. Not sure if I can fix it without requiring re-approval. I guess we'll find out.